request.go

```diff
@@ -203,6 +203,131 @@ func (exo *Client) RequestWithContext(ctx context.Context, request Command) (int
 	}
 }
 
+// AsyncRequest performs the given command
+func (exo *Client) AsyncRequest(request Command, callback WaitAsyncJobResultFunc) {
+	ctx, cancel := context.WithTimeout(context.Background(), exo.Timeout)
+	defer cancel()
+
+	switch request.(type) {
+	case syncCommand:
```

Comment:
`AsyncRequest` shouldn't accept that. It's maybe time to make `syncCommand` and `asyncCommand` public again.

Reply:
👍

Reply:
or only _async_